### PR TITLE
buildGoPackage: deps.json -> deps.nix in NIXON

### PR DESCRIPTION
**Pull request by kamilchm**
###### Motivation for this change

https://github.com/NixOS/nixpkgs/pull/17254#issuecomment-245295541
###### Things done
- [x] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
  - [x] NixOS
  - [ ] OS X
  - [ ] Linux
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by rushmorem** [comment]
The packages build fine but it seems like you forgot to update the docs. May I also suggest that we take this opportunity to mention the removal of `goPackages` in `nixos/doc/manual/release-notes/rl-1609.xml`.

**Comment by kamilchm** [comment]
Docs updated.
I don't know if I can find time for release notes before thursday :/

**Comment by rushmorem** [comment]
Had opened an issue for the release notes (https://github.com/NixOS/nixpkgs/issues/18496) but I see you have since added them. Thanks!

File: doc/languages-frameworks/go.xml [hunk 1]
```diff
@@ -24,7 +24,7 @@ deis = buildGoPackage rec {
     sha256 = "1qv9lxqx7m18029lj8cw3k7jngvxs4iciwrypdy0gd2nnghc68sw";
   };
 
-  goDeps = ./deps.json; <co xml:id='ex-buildGoPackage-3' />
+  goDeps = import ./deps.nix; <co xml:id='ex-buildGoPackage-3' />
```
Comment (rushmorem) [inline doc/languages-frameworks/go.xml]: Is `import` necessary here? I don't see it included in the packages you converted from `deps.json`.

File: nixos/doc/manual/release-notes/rl-1609.xml [hunk 2]
```diff
@@ -90,6 +90,13 @@ following incompatible changes:</para>
     Use <literal>security.audit.enable = true;</literal> to explicitly enable it.</para>
   </listitem>
 
+  <listitem>
+    <para><literal>goPackages</literal> was replaced with separated Go applications
+    in theirs <literal>nixpkgs</literal> categories. Each Go package uses its own
+    dependency set defined in NIXON format. There's also a new <literal>go2nix</literal>
+    tool introduced to generate Go package definition from its Go source automatically.
```
Comment (rushmorem) [inline nixos/doc/manual/release-notes/rl-1609.xml]: A closing `</para>` tag is missing here.

File: doc/languages-frameworks/go.xml [hunk 3]
```diff
@@ -56,7 +56,9 @@ the following arguments are of special significance to the function:
   <callout arearefs='ex-buildGoPackage-3'>
     <para>
       <varname>goDeps</varname> is where the Go dependencies of a Go program are listed
-      in a JSON format described below.
+      as a list of package source identified by Go import path.
+      It could be imported as a separate <varname>deps.nix</varname> file for
+      readability.  Dependency format is NIXON and is described below.
```
Comment (zimbatm) [inline doc/languages-frameworks/go.xml]: you and me know what NIXON means but not the reader.

> The dependency data structure is described below.

File: nixos/doc/manual/release-notes/rl-1609.xml [hunk 4]
```diff
@@ -90,6 +90,13 @@ following incompatible changes:</para>
     Use <literal>security.audit.enable = true;</literal> to explicitly enable it.</para>
   </listitem>
 
+  <listitem>
+    <para><literal>goPackages</literal> was replaced with separated Go applications
+    in theirs <literal>nixpkgs</literal> categories. Each Go package uses its own
+    dependency set defined in NIXON format. There's also a new <literal>go2nix</literal>
```
Comment (zimbatm) [inline nixos/doc/manual/release-notes/rl-1609.xml]: "define in nix", or we need to refer NIXON to the data-only subset of nix.

**Comment by zimbatm** [comment]
Looking good! Let me know if you want to change the NIXON references, otherwise I'll just merge it and add it to the 16.09 branch.

File: nixos/doc/manual/release-notes/rl-1609.xml (hunk 4 above)
Comment (rushmorem) [inline nixos/doc/manual/release-notes/rl-1609.xml]: `s/theirs/their/`

**Comment by kamilchm** [comment]
@zimbatm I replaced `NIXON` with `nix` as you suggested.

**Comment by kamilchm** [comment]
Telegraf needs to be converted https://github.com/NixOS/nixpkgs/pull/18437

**Comment by zimbatm** [comment]
Let me merge this and then update telegraph

**Comment by zimbatm** [comment]
Pushed as 914e0e594ca2d0fa5d456be208bf703d79f04fa3 on master. I squashed the commits to make it easier to cherry-pick onto the 16.09 branch.

**Comment by zimbatm** [comment]
Pushed as 9ab3dc25ceb7c604a072313492b9dca418bec87e on the release-16.09 branch. etcd had to be fixed.

**Comment by zimbatm** [comment]
I think everything's good now! Thanks a lot @kamilchm

**Pull request closed without merging**